src/nnet3/nnet-test-utils.cc

```diff
@@ -984,6 +984,12 @@ static void GenerateRandomComponentConfig(std::string *component_type,
       os << "new-column-order=" << buffer.str();
       break;
     }
+    case 21: {
+      *component_type = "PerElementOffsetComponent";
+      os << "dim=" << RandInt(1, 100)
+         << " learning-rate=" << learning_rate << " param-mean=0.0 param-stddev=1.0";
```

Comment:
On Mon, Nov 2, 2015 at 1:14 AM, hhadian <notifications@github.com> wrote:

> In src/nnet3/nnet-test-utils.cc
> <https://github.com/kaldi-asr/kaldi/pull/307#discussion_r43599487>:
>
> > @@ -984,6 +984,12 @@ static void GenerateRandomComponentConfig(std::string *component_type,
> >        os << "new-column-order=" << buffer.str();
> >        break;
> >      }
> > +    case 21: {
> > +      *component_type = "PerElementOffsetComponent";
> > +      os << "dim=" << RandInt(1, 100)
> > +         << " learning-rate=" << learning_rate << " param-mean=0.0 param-stddev=1.0";
>
> So I should also change the initialization that I have added here to
> "param-mean=0.0 param-stddev=0.0" ?
> Wouldn't that initialize offsets to all zero ? Is that ok for the test ?
>


Probably best to have nonzero parameters for the test.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/307/files#r43599487>.
>